QueryBuilder/Compilers/SqlServerCompiler.cs

```diff
@@ -60,8 +65,160 @@ public SqlServerCompiler()
             return result;
         }
 
+        private class SqlServerAggregatePercentileApproxColumn : SqlKata.AggregatePercentileApproxColumn
+        {
+            public SqlServerAggregatePercentileApproxColumn() : base() { }
+
+            public SqlServerAggregatePercentileApproxColumn(SqlServerAggregatePercentileApproxColumn other)
+                : base(other)
+            {
+                GroupByClauses = other.GroupByClauses;
+            }
+
+            public override AbstractClause Clone()
+            {
+                return new SqlServerAggregatePercentileApproxColumn(this);
+            }
+
+            public override string Compile(SqlResult ctx)
+            {
+                // percentile_cont(0.9) within group (order by "value") over (partition by ...)
+                var column = new Column { Name = Column }.Compile(ctx);
+                var partition = String.Join(", ", GroupByClauses.Select(clause => clause.Compile(ctx)));
+                partition = partition.Length > 0 ? "PARTITION BY " + partition : "";
+                return $"PERCENTILE_CONT({Percentile}) WITHIN GROUP(ORDER BY {column}) OVER({partition}) {ctx.Compiler.ColumnAsKeyword}{ctx.Compiler.WrapValue(Alias ?? Type)}";
+            }
+
+            public IEnumerable<Column> GroupByClauses { get; set; } = new List<Column>();
+        }
+
+        /**
+         * This transforms an APPROX_PERCENTILE column in a SELECT clause to a
+         * sub-query with an PERCENTILE_CONT column. Groupings in the original
+         * query are repeated in the PARTITION BY of the PERCENTILE_CONT
+         * function.
+         */
+        private Query TransformPercentileApprox(Query query)
+        {
+            Func<AbstractClause, bool> isApproxColumn = (clause) =>
+                clause.GetType() == typeof(SqlKata.AggregatePercentileApproxColumn);
+
+            if (query.Clauses.Where(isApproxColumn).Count() == 0)
+            {
+                return query;
+            }
+
+            var clauses = query.Clauses
+                .OrderBy(clause =>
+                {
+                    // to ensure selects are added to sub queries before they
+                    // are used in group-by or order-by clauses, so that we can
+                    // be sure to find them in the alias map.
+                    switch (clause.Component)
+                    {
+                        case "select": return 0;
+                        case "group": return 1;
+                        case "order": return 2;
+                        default: return 999;
+                    }
+                })
+                .ToList();
+
+            query.Clauses = new List<AbstractClause>();
+            const string alias = "percentile_approx";
```

Comment:
shouldn't this be prefixed with some `__` as its an internal identifier we are introducing? (for the subquery at least)

Reply:
also: also should be the base? column? `Type()`